apis/python/node/src/lib.rs

```diff
@@ -32,8 +33,12 @@ pub struct Node {
 #[pymethods]
 impl Node {
     #[new]
-    pub fn new() -> eyre::Result<Self> {
-        let (node, events) = DoraNode::init_from_env()?;
+    pub fn new(node_id: Option<String>) -> eyre::Result<Self> {
```

Comment:
So, having one API for both, make it possible to spawn the same node code using:

```bash
dora start dataflow.yaml
```

and:

```bash
dora start dataflow_dynamic.yaml

python plot.py
```

Making the code agnostic to being spawned dynamically or not.

I don't think that the code should be dependant on being spawned agnostically or not.

Reply:
I'm not against having multiple initialization function in Rust, but in Python, the main object class initialization tend to encapsulate all possible configuration as attributes.

Here are couple examples:
- ElasticSearch Client: https://www.elastic.co/guide/en/elasticsearch/client/python-api/current/connecting.html
- FastAPI server configuration: https://fastapi.tiangolo.com/reference/fastapi/#fastapi.FastAPI

The `new_flexible` class as you call it is usually the default configuration in Python.

It will be very strange in the pythonic world to have a very strict init function and a very loose init_flexible function.

By having agnostic code, I'm not saying that it is not going to be 100% fail-safe, but rather okay in most intuitive circumstance.

Reply:
Ok, good points!

I think we can avoid some of the behavior changes when we try the env-based initialization first and only try the ID-based initialization if it doesn't work. For the Rust side, I think a `new_flexible` function would be nice. Then the Python API can just call that function.